apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -1,32 +1,46 @@
+import type { IsomerComponent } from "@opengovsg/isomer-components"
 import { Box, Heading, HStack, Icon } from "@chakra-ui/react"
 import { Button, IconButton } from "@opengovsg/design-system-react"
 import { getComponentSchema } from "@opengovsg/isomer-components"
 import { BiDollar, BiX } from "react-icons/bi"
+import z from "zod"
 
 import { useEditorDrawerContext } from "~/contexts/EditorDrawerContext"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { trpc } from "~/utils/trpc"
 import FormBuilder from "./form-builder/FormBuilder"
 
-export default function ComplexEditorStateDrawer(): JSX.Element {
+const complexEditorSchema = z.object({
+  pageId: z.coerce.number(),
+  siteId: z.coerce.number(),
+})
```

Comment:
suggestion(blocking): we should use `editPageSchema` here instead

Reply:
sgtm, copied over the commit from `uat` @__@